src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_MultiBlockBase.java

```diff
@@ -294,6 +302,9 @@ public void loadNBTData(NBTTagCompound aNBT) {
         mEfficiency = aNBT.getInteger("mEfficiency");
         mPollution = aNBT.getInteger("mPollution");
         mRuntime = aNBT.getInteger("mRuntime");
+        if (supportsMachineModeSwitch()) {
+            machineMode = aNBT.getInteger("machineMode");
+        }
```

Comment:
Migration with `super.loadNBTData` won't work in your case, because this condition is wrong.
What happens with migration *before* `super.loadNBT` and current condition:
1) Migration properly loads machine mode from old tag.
2) This condition checks if multi-mode is supported (`true`).
3) This call overwrites machine mode from tag that *doesn't yet exist* and returns 0.

Whether machine supports multi-mode should only be checked on save. On load, you only need to check if tag exists:
```suggestion
        if (aNBT.hasKey("machineMode")) {
            machineMode = aNBT.getInteger("machineMode");
        }
```
And the whole logic will be:
1) Machine checks old tag and migrates to new system.
2) Machine calls `super.loadNBTData`.
3) if there's no `machineMode` tag (i. e., either it's old save or machine doesn't support multiple modes) then old (migrated) value is used. If `machineMode` tag is present then new tag is used.

Reply:
Why is it wrong to ignore the new tag if there's an old tag? Since we no longer save the old tag, doesn't that mean it will (on first load) set to the new tag, and on all subsequent loads the old tag will not exist? I made these changes but I'm definitely a bit in over my head